client/stmt.go

```diff
@@ -56,18 +57,33 @@ func (s *Stmt) Close() error {
 	return nil
 }
 
+// https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_com_stmt_execute.html
 func (s *Stmt) write(args ...interface{}) error {
 	paramsNum := s.params
 
 	if len(args) != paramsNum {
 		return fmt.Errorf("argument mismatch, need %d but got %d", s.params, len(args))
 	}
 
-	paramTypes := make([]byte, paramsNum<<1)
-	paramValues := make([][]byte, paramsNum)
+	if s.conn.includeLine >= 0 {
```

Comment:
Also only call `runtime.Caller` when we have set CLIENT_QUERY_ATTRIBUTES, to avoid the performance drop when the user enables `IncludeLine` but didn't enable CLIENT_QUERY_ATTRIBUTES